analyzers/src/SonarAnalyzer.CSharp/Rules/DisposableNotDisposed.cs

```diff
@@ -103,26 +100,24 @@ protected override void Initialize(SonarAnalysisContext context)
                         TrackAssignmentsToLocalsAndPrivateFields(typeDeclarationAndSemanticModel.SyntaxNode, typeDeclarationAndSemanticModel.SemanticModel, trackedNodesAndSymbols);
                     }
 
-                    if (trackedNodesAndSymbols.Any())
+                    if (!trackedNodesAndSymbols.Any())
```

Comment:
Was there an issue with too deep nesting of control flows? If not, I'd prefer the positive condition as before. The `if not condition return` is unnatural flow in the middle of some logic